pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionAssignmentTailer.java

```diff
@@ -34,15 +36,16 @@
         private final FunctionRuntimeManager functionRuntimeManager;
         private final Reader<byte[]> reader;
```

Comment:
Instead of byte[], do you think we could make it a typed reader?

Reply:
actually, it seems built-in `ProtobufSchema` couldn't deserialize `Function.Assignment` so, it seems it will require to add custom-schema and also, for deleted keys if topic is not compacted yet then consumer can consume message with empty-payload where consumer doesn't need to deserialize and AssignmentManager can directly delete assignment without deserializing message. so, i think keeping raw byte[] might be better choice.?